### PR TITLE
Add internal solution for Check Transcription (cf1056e)

### PR DESCRIPTION
**Pull request by ChuangSheep**
_If any of the below doesn't apply to this Pull Request, mark the checkbox as done._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions), which include the following: 
  - I understand that if it is clear that I have not attempted to follow these guidelines (ex. if I have not used tabs to indent), my PR will be closed.
  - If changes are requested, I will re-request the review after making them.

### CODE REVIEW

File: solutions/cf-1056E.mdx [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+			count_0++;
+		else
+			count_1++;
+	}
+
+	int pairs_count = 0;
+	// Brute force through all possible lengths of r0
+	for (int len_r0 = 1; len_r0 < t.size() / count_0; len_r0++) {
+		// This r0 is invalid if the length of r1 is no longer an integer
+		if ((t.size() - len_r0 * count_0) % count_1 != 0) continue;
```
Comment (SansPapyrus683) [inline solutions/cf-1056E.mdx:81]: braces

File: solutions/cf-1056E.mdx [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+	string s, t;
+	cin >> s >> t;
+	generate_hash(t);
+
+	int count_0 = 0;
+	int count_1 = 0;
+	for (char &c : s) {
+		if (c == '0')
+			count_0++;
+		else
```
Comment (SansPapyrus683) [inline solutions/cf-1056E.mdx:73]:
```suggestion
		else if (c == '1')
```
also braces

File: solutions/cf-1056E.mdx [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+	for (int i = 0; i < t.size(); i++) {
+		ull current_hash = ((rolling_hash.back() * base) % m + (ull)t[i]) % m;
+		rolling_hash.push_back(current_hash);
+		powers.push_back(powers.back() * base % m);
+	}
+}
+
+/**
+ * Get the hash value of the substring (a,b]
+ */
```
Comment (SansPapyrus683) [inline solutions/cf-1056E.mdx:56]:
```suggestion
/** Returns the hash value of substring (a, b] */
```
also why's the start exclusive but the end inclusive?
that's really weird

File: solutions/cf-1056E.mdx [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+
+ull m = 1e9 + 9;
+ull base = 9973;
+// Hashs and powers start with 1
+vector<ull> rolling_hash;
+vector<ull> powers;
+
+/**
+ * Calculate the prefix sums of the hashes for the given string t.
+ */
```
Comment (SansPapyrus683) [inline solutions/cf-1056E.mdx:43]:
```suggestion
/** Calculates the prefix sums of the hashes for the given string t. */
```

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: solutions/cf-1056E.mdx (hunk 3 above)
Comment (ChuangSheep) [reply solutions/cf-1056E.mdx:56]: well, for prefix sums its handy, but sure, I can change if u wish, just tell me

**Review by dongliu0426 — APPROVED** [review]

File: solutions/cf-1056E.mdx [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+	return ((m + rolling_hash[b]) - previous) % m;
+}
+
+int main() {
+	string s, t;
+	cin >> s >> t;
+	generate_hash(t);
+
+	int count_0 = 0;
+	int count_1 = 0;
```
Comment (SansPapyrus683) [inline solutions/cf-1056E.mdx:65]: `count0` and `count1` instead

**Review by SansPapyrus683 — APPROVED** [review]
yeah

**Pull request merged**